module-4/backend/week_three.md

```diff
@@ -5,15 +5,46 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is Node?
+
+Node is an asynchronous JavaScript runtime built on Chrome's V8 JS engine used to run server-side code.
+
 2. What is Express? Why is Express similar to in the Ruby world?
+
+Express is a JS framework that's hosted on the Node.js runtime environment. It creates handlers for HTTP requests at different routes. It's similar to Rails in the Ruby world.
+
 3. How do we setup a route when creating an API with Node and Express? Please provide a code snippet.
+
+```
+app.get('/', function(request, response) {
+  response.send('Hello World!')
+})
+```
+
 4. What do we use Knex for?
+
+Knex is used to build SQL database queries in a quick and easy way.
+
 5. How could you organize your code to follow the MVC design pattern in your Quantified Self project?
+
+SQL queries can be extracted into their own model file, the middleware functions can be extracted into their own controller file and the view renderings can be extracted into their own file. 
+
 6. How do you execute raw SQL in node?
+
+To execute raw SQL in node, use Knex and run database.raw() with the SQL query included within the parentheses.
+
 7. What are some advantages to having your front end and back end code live in separate applications? What are some disadvantages?
 
+The advantages of separating front end from back end code is that they can be structured modularly with differing development, testing and deployment technologies along with differing development paces. It also allows for more agile creation and future adjustments. The disadvantages include utilizing more time and resources to build out the separate technologies and testing.
+
 #### Review  
 
 8. Describe DNS.
+
+DNS is a Domain Name Server which translates domain names to IP addresses from a database.
+
 9. What does writing clean code mean to you?
+
+Clean code means writing code that is easily organized and structured. Methods/functions are limited to a reasonable amount of lines of code and arguments.
+
 10. If you were building an application that models hotels, what classes would you need? What classes would be responsible for what functionality?
+Classes: Rooms, Guests, Reservations, Bill, Room Service, Wait Staff, Management, Housekeeping
```

Comment:
Would love to see this a little more fleshed out. What are the classes' responsibilities? How would they interact with each other?